### ISSUE TITLE
Ignore config.json "ExperimentalDefaultChannels" & "Town Square" channel

### ISSUE THREAD

**Issue by baek-sang** [mattermost/mattermost-server#17970]
#### Summary
Restart mattermost after set config.json file 
```
"ExperimentalDefaultChannels": [
            "ex1",
            "ex2"
        ]
```
but create new team, "Town Square" channel created

#### Steps to reproduce
Server 5.37.0 self-host E0

#### Expected behavior
- Delete or change default channel.
- Don't create "Town Square" channel.

#### Observed behavior (that appears unintentional) 
![ccccc](https://user-images.githubusercontent.com/69100154/126377470-27b49e20-c113-429d-93c5-8c937481cb81.png)
Two channel name is empty.
Why "Town Square" auto created? no delete, no chagne default channel? no change url?

#### Possible fixes
If you can, link to the line of code that might be responsible for the problem

**Comment by amyblais** [comment]
As per [the documentation](https://docs.mattermost.com/configure/configuration-settings.html#default-channels-experimental), "Note that even if town-square is not listed, every user is added to that channel after joining a new team."

However, with the other default channels being created as empty, @hmhealey @lindalumitchell are you able to reproduce that and is it a regression?

**Comment by baek-sang** [comment]
@amyblais How to leave(delete) Town Square channel? or change url.

**Comment by hmhealey** [comment]
@baek-sang Unfortunately, it's not possible to leave/delete the Town Square. There are parts of Mattermost that rely on it existing, so even if you were to find a way to leave/delete it, it would cause parts of the app to stop working.

**Comment by amyblais** [comment]
@hmhealey Thanks, is it a bug that the other default channels are created as empty as shown in the screenshot https://github.com/mattermost/mattermost-server/issues/17970#issue-948936877?

**Comment by lindalumitchell** [comment]
We don't have a regular test case for this experimental setting that I could find. I'll do some more investigation to see if I can find more info / reproduce the empty channels.

**Comment by lindalumitchell** [comment]
I found some [original test steps from when this was added](https://mattermost.atlassian.net/browse/MM-11180). It does appear that there is a bug here, but not exactly as originally described. How it should work:
1. Create new public channels PublicA and PublicB on TeamOne
2. Set Team Settings.ExperimentalDefaultChannels to ["publica", "publicb"]
3. Add new user to TeamOne
Expected: New user is member of Town Square plus channels PublicA and PublicB on TeamOne. The public channels that will be default must be created on the team *before* the new user is added.

Then: If you create a new team, TeamTwo, but _do not create the channels first_ and add a new user, that user will only be a member of Town Square on that team. But it should not create empty channels in any case. 

@baek-sang if you go through the steps above, **creating the channels on the team before adding a new user**, do the channel names still appear empty?
